prefect_dbt/cli/credentials.py

```diff
@@ -0,0 +1,84 @@
+"""Module containing credentials for interacting with dbt CLI"""
+from dataclasses import dataclass
+
+
+@dataclass
+class DbtCliCredentials:
```

Comment:
I would recommend renaming this class to `DbtCliProfile` and update the attributes to mirror the structure of a dbt CLI `profiles.yml` (reference here: https://docs.getdbt.com/reference/profiles.yml). I don't think we should have `user`, `password`, `host`, or `port`, because not all dbt CLI profiles need those values.

Concretely, the args on this class would be `name`, `target`, and `profile_config`. `profile_config` would need to be a `Dict` type to make sure it can be input in the Prefect UI.

Reply:
Can we keep `Credentials` rather than `Profile`? I think all our collections so far have Credentials?

Reply:
Maybe we can have specific Profiles that inherit from DbtCliProfile then, e.g. SnowflakeDbtCliProfile.

Reply:
I imagine creating only a few popular ones, like snowflake, gcp, and ask for contributions but I guess can be in another PR.

Reply:
+1 to a generic one, even though I believe it's not as bad as 30+ since currently dbt Labs supports only 5 Postgres, Snowflake, BigQuery, Spark and Redshift, but the number of vendor- + community-contributed ones keeps growing so even 50+ may easily become a reality in a near future